modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/sql/GridSqlQueryParser.java

```diff
@@ -1323,9 +1293,64 @@ private GridSqlCreateTable parseCreateTable(CreateTable createTbl) {
                 res.affinityKey(pkCols0.iterator().next());
         }
 
+        if (implicitPk) {
+            log.warning("No PRIMARY KEY defined for table '" + data.tableName + "'. " +
+                "An auto-generated implicit primary key will be created.");;
+        }
+
         return res;
     }
 
+    /**
+     * @param createTbl {@code CREATE TABLE} statement.
+     * @param cols Table columns.
+     * @return Primary key column names.
+     */
+    private LinkedHashSet<String> parseConstraints(CreateTable createTbl, Map<String, GridSqlColumn> cols) {
+        List<DefineCommand> constraints = CREATE_TABLE_CONSTRAINTS.get(createTbl);
+
+        if (constraints.size() > 1) {
+            throw new IgniteSQLException("Too many constraints - only PRIMARY KEY is supported for CREATE TABLE",
+                IgniteQueryErrorCode.UNSUPPORTED_OPERATION);
+        }
+
+        DefineCommand constraint = constraints.get(0);
+
+        if (!(constraint instanceof AlterTableAddConstraint)) {
+            throw new IgniteSQLException("Unsupported type of constraint for CREATE TABLE - only PRIMARY KEY " +
+                "is supported", IgniteQueryErrorCode.UNSUPPORTED_OPERATION);
+        }
+
+        AlterTableAddConstraint alterTbl = (AlterTableAddConstraint)constraint;
+
+        if (alterTbl.getType() != Command.ALTER_TABLE_ADD_CONSTRAINT_PRIMARY_KEY) {
+            throw new IgniteSQLException("Unsupported type of constraint for CREATE TABLE - only PRIMARY KEY " +
+                "is supported", IgniteQueryErrorCode.UNSUPPORTED_OPERATION);
+        }
+
+        AlterTableAddConstraint pk = CREATE_TABLE_PK.get(createTbl);
+
+        IndexColumn[] pkIdxCols = ALTER_TABLE_ADD_CONSTRAINT_INDEX_COLUMNS.get(pk);
```

Comment:
I understand that this is just a copy-pasted lines, but... If we are deriving pk once again in line 1331, then we don't need lines 1319-1324 (`getType` can be accessed on DefineCommand directly), or we can reuse already validated `alterTbl` constraint

Reply:
Thanks, removed redundant check (1319-1324 lines).